astropy/units/__init__.py

```diff
@@ -40,20 +40,23 @@
 from .equivalencies import *
 from .function import *
 
-__all__ = []
-__all__ += core.__all__
-__all__ += quantity.__all__
-__all__ += decorators.__all__
-__all__ += structured.__all__
-__all__ += si.__all__
-__all__ += cgs.__all__
-__all__ += astrophys.__all__
-__all__ += physical.__all__
-__all__ += misc.__all__
-__all__ += photometric.__all__
-__all__ += structured.__all__
-__all__ += equivalencies.__all__
-__all__ += function.__all__
+# See __init__.pyi for static-compatible __all__ definition.
+# This __all__ and the one in __init__.pyi should be kept in sync!
+__all__: list[str] = sorted(  # noqa: PLE0605
+    core.__all__
+    + quantity.__all__
+    + decorators.__all__
+    + structured.__all__
+    + si.__all__
+    + cgs.__all__
+    + astrophys.__all__
+    + physical.__all__
+    + misc.__all__
+    + photometric.__all__
+    + structured.__all__
+    + equivalencies.__all__
+    + function.__all__
+)
```

Comment:
Are we sure that static analyzers would have a problem with
```
__all__.sort()
```

Reply:
Pretty sure. That's not one of the allowed operations.
https://github.com/python/typing/blob/master/docs/source/libraries.rst#library-interface-public-and-private-symbols

Reply:
If we use https://github.com/scientific-python/lazy_loader then this would become

import lazy_loader as lazy
__getattr__, __dir__, __all__ = lazy.attach_stub(__name__, "subpackages.pyi")
__all__ = __all__.sort()

Reply:
Sounds like it is worth another upstream ticket - in-place sorting does nothing to the contents, so why couldn't one do it?

Alternatively, maybe sphinx has an option to alphabetize? Actually, `automodapi` is under our control, so we can add it ourselves...

Reply:
> Alternatively, maybe sphinx has an option to alphabetize? Actually, automodapi is under our control, so we can add it ourselves.

https://github.com/astropy/sphinx-automodapi/pull/182